frame/nomination-pools/src/lib.rs

```diff
@@ -899,6 +899,60 @@ impl<T: Config> BondedPool<T> {
 		Ok(points_issued)
 	}
 
+	/// Bond all of the pending_rewards from `member` into this pool.
+	///
+	/// Similar to `try_bond_funds`, but doesn't transfer funds from an account, but instead
+	/// transfers the rewards directly from the member's reward pool to the `bonded_account`.
+	///
+	/// Returns `Ok((points_issues, bonded))`, `Err` otherwise.
+	fn try_bond_funds_from_rewards(
```

Comment:
Can you revisit this function? most of it seems like duplicate code from other functions to me.

Reply:
Some of the comments were duplicated, and I changed them, but other than that I don't see problems with the logic. Could you please be a bit more specific?

Reply:
Maybe I could change the ```do_reward_payout``` function so that we can use it from the ```try_bond_funds_from_rewards``` also. Did you think of something like that?